internal/worker_versioning_rules.go

```diff
@@ -419,8 +422,10 @@ func (r *VersioningAssignmentRule) validateRule() error {
 	}
 	switch ramp := r.Ramp.(type) {
 	case *VersioningRampByPercentage:
-		if err := ramp.validateRamp(); err != nil {
-			return err
+		if ramp != nil {
```

Comment:
Can me move the `nil` check out of the switch so when we add more `Ramp` types we don't need to duplicate this check?

Reply:
`Ramp` is an interface and to know that a non-nil interface contains a nil pointer I need to type cast. Can't think of a way to do that without repeating the type cast for each concrete type, kind of defeating the purpose of lifting the check...